### ISSUE TITLE
Item add/remove/change/move animations do not work

### ISSUE THREAD

**Issue by teshlya** [saket/InboxRecyclerView#16]
Hello. How remove item from recyclerview?

`     @Override
            public void onSwiped(RecyclerView.ViewHolder viewHolder, int swipeDir) {
                int position = viewHolder.getAdapterPosition();
                articles.remove(position);
                notifyDataSetChanged();
            }`

If this code, then I have white empty space. Video - https://drive.google.com/file/d/1nnJ3AXXailegOfxaYb-cykoOs72A3oZ0/view

`     @Override
            public void onSwiped(RecyclerView.ViewHolder viewHolder, int swipeDir) {
                int position = viewHolder.getAdapterPosition();
                articles.remove(position);
                notifyItemRemoved(position);
            }`

If this code, then I have error on position - 0. Video - https://drive.google.com/file/d/13W2CnKzdIYUlqes5HojwMwIC-57kKxSo/view

`java.lang.IndexOutOfBoundsException: Inconsistency detected. Invalid item position 0(offset:-1).state:25 me.saket.inboxrecyclerview.InboxRecyclerView{48c7ea0 VFED..... .F....ID 0,0-1440,2448 #7f090038 app:id/community_recycler_view}, adapter:teshlya.com.reddit.adapter.CommunityAdapter@239767f, layout:androidx.recyclerview.widget.LinearLayoutManager@9cc564c, context:teshlya.com.reddit.screen.ArticleActivity@9771922
        at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5923)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5858)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5854)
        at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2230)
        at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1557)
        at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1517)
        at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:622)
        at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep1(RecyclerView.java:3875)`

**Comment by teshlya** [comment]
`@Override public void onSwiped(RecyclerView.ViewHolder viewHolder, int swipeDir) { int position = viewHolder.getAdapterPosition(); articles.remove(position); notifyDataSetChanged(); }`

This work good with RecyclerView, but with InboxRecyclerView I have empty white space - https://drive.google.com/file/d/1nnJ3AXXailegOfxaYb-cykoOs72A3oZ0/view

**Comment by saket** [comment]
Oh hey I know this bug. I've already pushed a fix ([commit](https://github.com/saket/InboxRecyclerView/commit/354e5c119c6661bedff67a16776dd66d0c8bde04)), but you will unfortunately have to wait for the next release. I'll make one in a day or two.

If it's blocking you then you can get the latest snapshot:

```groovy
repositories {
  maven { url = "https://oss.sonatype.org/content/repositories/snapshots/" }
}

dependencies {
  implementation 'me.saket:inboxrecyclerview:1.0.0-SNAPSHOT'
}
```

**Comment by saket** [comment]
Should be fixed in [v1.0.0-rc1](https://github.com/saket/InboxRecyclerView/releases/tag/v1.0.0-rc1). Please feel free to reopen this if the issue is still present.
